test/Microsoft.AspNet.StaticFiles.Tests/StaticFileContextTest.cs

```diff
@@ -49,13 +51,54 @@ public void LookupFileInfo_ReturnsTrue_IfFileExists()
             var context = new StaticFileContext(httpContext, options, pathString, NullLogger.Instance);
 
             // Act
-            context.ValidatePath();
+            context.MatchPath();
             var result = context.LookupFileInfo();
 
             // Assert
             Assert.True(result);
         }
 
+        [Fact]
+        public void ValidatePath_ReturnsFalse_IfPathContainsInvalidCharacters()
+        {
+            // Test whole set of invalid chars in one test as testing indivually may be slow due to number of invalid path chars on Windows
+            var invalidPath = new string(Path.GetInvalidPathChars());
+
+            // Arrange
+            var options = new StaticFileOptions();
+            options.FileProvider = new TestFileProvider();
+            var pathString = new PathString("/test");
+            var httpContext = new DefaultHttpContext();
+            httpContext.Request.Path = new PathString($"/test/{invalidPath}.txt");
+            var context = new StaticFileContext(httpContext, options, PathString.Empty, NullLogger.Instance);
+
+            // Act
+            context.MatchPath();
```

Comment:
The call to MatchPath() is required to populate the _subPath as part of the StaticFilesContext. But the entire test could go, it's pretty much already covered as part of the middleware test. I just thought I'd have add a unit test for the changes to StaticFilesContext as well as the integration test with the middleware and starting up a test server.